static/app/views/profiling/profileDetails/hooks/useSortableColumn.ts

```diff
@@ -0,0 +1,87 @@
+import {useCallback, useMemo} from 'react';
+
+import {GridColumnSortBy} from 'sentry/components/gridEditable';
+
+import {useQuerystringState} from './useQuerystringState';
+
+export function useSortableColumns<T extends string>(options: {
+  defaultSort: GridColumnSortBy<T>;
+  querystringKey: string;
+  sortableColumns: readonly string[];
+}) {
+  const {sortableColumns, querystringKey, defaultSort} = options;
+  const [queryStringState, _, createLocationDescriptor] = useQuerystringState({
+    key: querystringKey,
+  });
+  const currentSort = useMemo<GridColumnSortBy<T>>(() => {
+    let key = queryStringState ?? '';
+
+    const isDesc = key[0] === '-';
+    if (isDesc) {
+      key = key.slice(1);
+    }
+
+    if (!key || !sortableColumns.includes(key as T)) {
+      return defaultSort;
+    }
+
+    return {
+      key,
+      order: isDesc ? 'desc' : 'asc',
+    } as GridColumnSortBy<T>;
+  }, [sortableColumns, defaultSort, queryStringState]);
+
+  const generateSortLink = useCallback(
+    (column: T) => {
+      if (!sortableColumns.includes(column)) {
+        return () => undefined;
+      }
+      if (!currentSort) {
+        return () => createLocationDescriptor(column);
+      }
+
+      const direction =
+        currentSort.key !== column
+          ? 'desc'
+          : currentSort.order === 'desc'
+          ? 'asc'
+          : 'desc';
```

Comment:
Can we simplify this a little with something like
```suggestion
      const direction =
        currentSort.key === column && currentSort.order === 'desc'
          ? 'asc'
          : 'desc';
```